test/end2end_test.go

```diff
@@ -4257,29 +4257,34 @@ func TestClientResourceExhaustedCancelFullDuplex(t *testing.T) {
 func testClientResourceExhaustedCancelFullDuplex(t *testing.T, e env) {
 	te := newTest(t, e)
 	recvErr := make(chan error, 1)
-	timeoutErr := errors.New("1s timeout reached")
 	ts := &funcServer{fullDuplexCall: func(stream testpb.TestService_FullDuplexCallServer) error {
+		defer close(recvErr)
 		_, err := stream.Recv()
 		if err != nil {
-			t.Fatalf("expecting <nil> error, got %v", err)
+			return err
```

Comment:
A little nitty, but to be safe, this and the `return` below should return a specific new status error that is not `ResourceExhausted`.